thirdeye-server/src/main/java/ai/startree/thirdeye/resources/CrudResource.java

```diff
@@ -206,11 +214,36 @@ public boolean hasAccess(final ThirdEyePrincipal principal, final AbstractDTO dt
 
   public void ensureHasAccess(final ThirdEyePrincipal principal, final AbstractDTO dto,
```

Comment:
Is there a way we can move all of this logic outside? meaning we create a generic service class which has all these methods and instances to decide it all and let `CrudResource` know? This is a lot of dedicated code for auth which could be containerized? thoughts?

Reply:
🤔 Ya, I agree it needs it's own class. I'm starting to think a wrapper on top of the data managers might be the way to go. For example, when CrudResource calls dataManager.findById(), it's an auth wrapper which calls realManager.findByID() and does the access control check.

This would isolate all the auth code from CrudResource and also probably save me from missing an endpoint.

Reply:
Decided not to go w/ wrapper ... was way to clunky haha. I made a class AuthorizationManager to handle the this code.